libs/commons/src/main/kotlin/uk/gov/justice/digital/hmpps/retry/Retry.kt

```diff
@@ -1,13 +1,24 @@
 package uk.gov.justice.digital.hmpps.retry
 
-fun <T> retry(maxRetries: Int, code: () -> T): T {
-    var throwable: Throwable? = null
-    (1..maxRetries).forEach { _ ->
+import kotlin.reflect.KClass
+
+fun <T> retry(maxRetries: Int, exceptions: List<KClass<out Exception>> = listOf(Exception::class), code: () -> T): T {
+    var throwable: Throwable?
+    (1..maxRetries).forEach { count ->
```

Comment:
Do we need any kind of delay or back-off between retries?

Reply:
added a delay and using Feign retry stuff